components/codec/src/byte.rs

```diff
@@ -98,6 +98,53 @@ impl MemComparableByteCodec {
         }
     }
 
+    /// Encodes all bytes in the `src` in ascending memory-comparable format in place.
+    /// If the capacity of `src` is less than the encoded length, `src` will `reserve` the
+    /// minimum sufficient capacity.
+    ///
+    /// Returns the number of bytes encoded.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the encoded length overflows usize.
+    pub fn encode_all_in_place(src: &mut [u8]) -> usize {
+        // Refer: https://github.com/facebook/mysql-5.6/wiki/MyRocks-record-format#memcomparable-format
+        let src_len = src.len();
+        let dest_len = Self::encoded_len(src_len);
+
+        // There must be 0 or more zero padding groups and 1 non-zero padding groups
+        // in the output.
+        let zero_padding_groups = src_len / MEMCMP_GROUP_SIZE;
+
+        // To avoid overwriting un-copied bytes, we write the last group first,
+        // then write zero padding groups backward.
+
+        // First, write the last group, which should never be zero padding.
+        let remaining_size = src_len - MEMCMP_GROUP_SIZE * zero_padding_groups;
```

Comment:
maybe call it last_group_size? Since it is not the "remaining" part  in the context of this function.